### PR TITLE
feat(Modal): export new Modal component to /next

### PR DESCRIPTION
**Pull request by YossiSaadi**
- **refactor**: Move Modal folder to be under ./Modal
- **docs**: Remove 'internal' tag from stories, so stories and mdxs would be available in the public Storybook
- **feat**: Export new Modal to /next

https://monday.monday.com/boards/3532714909/pulses/7360832642
https://monday.monday.com/boards/3532714909/pulses/7360977361
https://monday.monday.com/boards/3532714909/pulses/7989661376
https://monday.monday.com/boards/3532714909/pulses/8010109228

### CODE REVIEW

File: packages/core/src/next.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from "./components/next";
```
Comment (talkor) [inline packages/core/src/next.ts:1]: Why this again?

File: packages/core/package.json [hunk 2]
```diff
@@ -36,9 +36,9 @@
       "default": "./dist/src/tests/testIds.js"
     },
     "./next": {
-      "types": "./dist/components/next/next.d.ts",
-      "import": "./dist/src/components/next/next.js",
-      "default": "./dist/src/components/next/next.js"
+      "types": "./dist/next.d.ts",
+      "import": "./dist/src/next.js",
+      "default": "./dist/src/next.js"
```
Comment (talkor) [inline packages/core/package.json:41]: Why this?

**Review by talkor — CHANGES_REQUESTED** [review]

File: packages/core/src/components/Modal/layouts/ModalBasicLayout/__stories__/ModalBasicLayout.stories.tsx [hunk 3]
```diff
@@ -25,12 +25,14 @@ const metaSettings = createStoryMetaSettingsDecorator({
 });
 
 export default {
-  title: "Internal/Components/Modal [New]/Basic modal",
+  title: "Components/Modal [New]/Basic modal",
```
Comment (talkor) [inline packages/core/src/components/Modal/layouts/ModalBasicLayout/__stories__/ModalBasicLayout.stories.tsx:28]: Feels a bit weird title no?

Comment (YossiSaadi) [reply packages/core/src/components/Modal/layouts/ModalBasicLayout/__stories__/ModalBasicLayout.stories.tsx:28]: why?

File: packages/core/package.json (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/package.json:41]: commented above

File: packages/core/src/next.ts (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/src/next.ts:1]: yet

Comment (talkor) [reply packages/core/src/next.ts:1]: Feels a bit like preparation for AC but OK :)

**Review by talkor — DISMISSED** [review]
💯

File: packages/core/src/next.ts (hunk 1 above)
Comment (YossiSaadi) [reply packages/core/src/next.ts]: I removed the preparation for AC, you are right 🙃

**Review by talkor — APPROVED** [review]
Exciting! 🤩 
And damn that's one big component with many files

**Pull request merged**